### ISSUE TITLE
Check that GZI index provided if using a gzipped FASTA

### ISSUE THREAD

**Issue by pontushojer** [nf-core/sarek#1741]
### Description of feature

I started a run providing a custom reference in the form of a gzipped FASTA. The run from FASTQs started normally and did not run into any errors until the MarkDuplicates step. I had missed copying an index file (*.fasta.gz.gzi) to the same folder as the FASTA which caused the step to fail just before finishing 🤦 , see the error message below.

```
  [Thu Nov 28 20:40:47 GMT 2024] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 134.77 minutes.
  Runtime.totalMemory()=285212672
  [E::bgzf_index_load] Error opening GRCh38_GIABv3_no_alt_analysis_set_maskedGRC_decoys_MAP2K3_KMT2C_KCNJ18.fasta.gz.gzi : No such file or directory
  [E::bgzf_open_ref] Unable to load .gzi index 'GRCh38_GIABv3_no_alt_analysis_set_maskedGRC_decoys_MAP2K3_KMT2C_KCNJ18.fasta.gz.gzi'
  [E::refs_load_fai] Failed to open reference file 'GRCh38_GIABv3_no_alt_analysis_set_maskedGRC_decoys_MAP2K3_KMT2C_KCNJ18.fasta.gz'
  [E::hts_open_format] Failed to open file "OPM2.md.cram" : Invalid argument
  samtools view: failed to open "OPM2.md.cram" for writing: Invalid argument
```

This is the relevant part of my parameter file
```
fasta: /proj/ngi2016004/private/strategic_proj/SR_23_02_Element_vs_Illumina/resources/GRCh38_GIABv3/GRCh38_GIABv3_no_alt_analysis_set_maskedGRC_decoys_MAP2K3_KMT2C_KCNJ18.fasta.gz
fasta_fai: /proj/ngi2016004/private/strategic_proj/SR_23_02_Element_vs_Illumina/resources/GRCh38_GIABv3/GRCh38_GIABv3_no_alt_analysis_set_maskedGRC_decoys_MAP2K3_KMT2C_KCNJ18.fasta.gz.fai
igenomes_ignore: true
bwa: /proj/ngi2016004/private/strategic_proj/SR_23_02_Element_vs_Illumina/resources/GRCh38_GIABv3/BWAIndex
```

It would be great if there could be a parameter check on start so that when a gzipped fasta `*.fasta.gz` is provided a corresponding index `*.fasta.gz.gzi` should be present. 

Some other considerations if this is hard to implement:
- This `.gzi` file could be generated automatically if it is missing as part of the `build` process. 
- The MarkDuplicates step should have a check that if a gzipped FASTA is provided an index file is also present before starting the step, making the run fail earlier.

**Comment by sbdaxia** [comment]
I'm also running into the same issue. I'm using Sarek v3.5.0.

I tried a run with my bgzipped custom reference `.fa.gz`, `.fa.gz.fai`, and `.fa.gz.gzi` files in the same folder and set the `--fasta` and `--fasta_fai` parameter to point to the respecetive files. There was no parameter for specifying the `.fa.gz.gzi` file. I have then tried to do another run only specifying the `--fasta` parameter. In both cases, I ran into the same error.

I'm guessing the pipeline doesn't support using zipped custom reference as input?

**Comment by FriederikeHanssen** [comment]
We have not specifically implemented gzi support, so it is quite possible that it won't work out of the box. Looks like the tool needs some particular setting to deal with it? Do you by chance know what the tool needs?

**Comment by sbdaxia** [comment]
You are absolutely right. I just unzipped the gzipped reference genome file and supplied the pipeline with the `.fa` and `.fa.fai` file and it ran successfully.

Based on the error code, it seems like if a `.fa.gz` file is supplied, a flag specifies the location of the `fa.gz.gzi` file also needs to be provided. Even though my `fa.gz.gzi` file was in the same directory as the ".fa.gz" file, it did not automatically detect it.

Here is the relevant bit of error I got. Hope that it is helpful!

```
  [E::bgzf_index_load] Error opening Mus_musculus.GRCm39.dna_sm.primary_assembly_wPE7.fa.gz.gzi : No such file or directory
  [E::bgzf_open_ref] Unable to load .gzi index 'Mus_musculus.GRCm39.dna_sm.primary_assembly_wPE7.fa.gz.gzi'
  [E::refs_load_fai] Failed to open reference file 'Mus_musculus.GRCm39.dna_sm.primary_assembly_wPE7.fa.gz'
  [E::hts_open_format] Failed to open file "normal.md.cram" : Invalid argument
  samtools view: failed to open "normal.md.cram" for writing: Invalid argument
```

**Comment by FriederikeHanssen** [comment]
ok interesting. I suppose if people have the files in this format more frequently we can support this. one weird thing you could try, is using a wildcard for the fai file, so that nextflow stages both. I don't think there is any process that directly references the fai file (so if both end up in the same directory it might not be a problem).
